drivers/usb/device/usb_dc_sam_usbc.c

```diff
@@ -0,0 +1,1160 @@
+		if (!dev_data.ep_data[i].is_configured || i == ep_idx) {
+			continue;
+		}
+
+		desc_mem_alloc += dev_data.ep_data[i].mps_x2
+				? dev_data.ep_data[i].mps * 2
+				: dev_data.ep_data[i].mps;
+	}
+
+	if ((desc_mem_alloc + mps) > USBC_RAM_SIZE) {
+		memset(&dev_data.ep_data[ep_idx], 0,
+		       sizeof(struct usb_device_ep_data));
+
+		return -ENOMEM;
+	}
+
+	/*
+	 * Allocate the memory. This part is a bit tricky as memory can only
+	 * be allocated if all above endpoints are disabled and not allocated.
+	 * Loop backward through the above endpoints, disable them if they are
+	 * enabled, deallocate their memory if needed. Then loop again through
+	 * all the above endpoints to allocate and enabled them.
+	 */
+	for (int i = NUM_OF_EP_MAX - 1; i >= ep_idx; i--) {
+		ep_enabled[i] = usb_dc_ep_is_enabled(i);
+
+		if (ep_enabled[i]) {
+			LOG_DBG("Temporary disable ep idx 0x%02x", i);
+			usb_dc_ep_disable(i);
+		}
+	}
+
+	desc_mem_alloc = 0U;
+	for (int i = 0; i < ep_idx; i++) {
+		if (!dev_data.ep_data[i].is_configured) {
+			continue;
+		}
+
+		desc_mem_alloc += dev_data.ep_data[i].mps_x2
+				? dev_data.ep_data[i].mps * 2
+				: dev_data.ep_data[i].mps;
+	}
+
+	ep_desc_bk = ((struct sam_usbc_desc_table_t *) &dev_desc)
+		   + (ep_idx * 2);
+	for (int i = ep_idx; i < NUM_OF_EP_MAX; i++) {
+		if (!dev_data.ep_data[i].is_configured && (i != ep_idx)) {
+			ep_desc_bk += 2;
+			continue;
+		}
+
+		/* Alloc bank 0 */
+		ep_desc_bk->ep_pipe_addr = ((uint8_t *) USBC_RAM_ADDR)
+					 + desc_mem_alloc;
+		ep_desc_bk->sizes = 0;
+		ep_desc_bk->bk_ctrl_stat = 0;
+		ep_desc_bk->ep_ctrl_stat = 0;
+		ep_desc_bk++;
+
+		/**
+		 * Alloc bank 1
+		 *
+		 * if dual bank,
+		 * then ep_pipe_addr[1] = ep_pipe_addr[0] address + mps size
+		 * else ep_pipe_addr[1] = ep_pipe_addr[0] address
+		 */
+		ep_desc_bk->ep_pipe_addr = ((uint8_t *) USBC_RAM_ADDR)
+					 + desc_mem_alloc
+					 + (dev_data.ep_data[i].mps_x2
+					 ?  dev_data.ep_data[i].mps
+					 :  0);
+		ep_desc_bk->sizes = 0;
+		ep_desc_bk->bk_ctrl_stat = 0;
+		ep_desc_bk->ep_ctrl_stat = 0;
+		ep_desc_bk++;
+
+		desc_mem_alloc += dev_data.ep_data[i].mps_x2
+				? dev_data.ep_data[i].mps * 2
+				: dev_data.ep_data[i].mps;
+	}
+
+	ep_enabled[ep_idx] = false;
+	for (int i = ep_idx; i < NUM_OF_EP_MAX; i++) {
+		if (ep_enabled[i]) {
+			usb_dc_ep_enable(i);
+		}
+	}
+
+	return 0;
+}
+
+static void usb_dc_ep_enable_interrupts(uint8_t ep_idx)
+{
+	if (ep_idx == 0U) {
+		/* Control endpoint: enable SETUP */
+		regs->UECONSET[ep_idx] = USBC_UECON0SET_RXSTPES;
+
+	} else if (regs->UECFG[ep_idx] & USBC_UECFG0_EPDIR_IN) {
+		/* TX - IN direction: acknowledge FIFO empty interrupt */
+		regs->UESTACLR[ep_idx] = USBC_UESTA0CLR_TXINIC;
+		regs->UECONSET[ep_idx] = USBC_UECON0SET_TXINES;
+
+	} else {
+		/* RX - OUT direction */
+		regs->UECONSET[ep_idx] = USBC_UECON0SET_RXOUTES;
+	}
+}
+
+static void usb_dc_ep_isr_sta(uint8_t ep_idx)
+{
+	uint32_t sr = regs->UESTA[ep_idx];
+
+#if defined(CONFIG_USB_DRIVER_LOG_LEVEL_DBG)
```

Comment:
Please add usb_dc_ep_isr_sta_dbg() inside `#if defined(CONFIG_USB_DRIVER_LOG_LEVEL_DBG)` at L:25, similar to usb_dc_sam_usbc_clean_sta_dbg()